api/src/bigquerytest/java/org/pmiops/workbench/api/CohortBuilderControllerBQTest.java

```diff
@@ -685,18 +663,201 @@ public void countSubjectsBPMessageException() throws Exception {
+    searchGroupIcd10Child.temporalGroup(1);
+    searchGroup = new SearchGroup()
+      .items(Arrays.asList(searchGroupIcd9Child, searchGroupSnomed, searchGroupIcd10Child))
+      .temporal(true)
+      .mention(TemporalMention.ANY_MENTION.name())
+      .time(TemporalTime.X_DAYS_AFTER.name())
+      .timeValue(5L);
+    searchRequest = new SearchRequest().includes(Arrays.asList(searchGroup));
+    //matches searchGroupSnomed and searchGroupIcd10Child
+    assertParticipants(controller.countParticipants(cdrVersion.getCdrVersionId(), searchRequest), 1);
+
+    //Any Mention Of (ICD10Parent(modifier) or CPT) within 5 Days of visit
+    //First temporal group
+    searchGroupIcd10Parent.temporalGroup(0);
+    searchGroupCpt.temporalGroup(0);
+    //Second temporal group
+    searchGroupVisit.temporalGroup(1);
+    searchGroup = new SearchGroup()
+      .items(Arrays.asList(searchGroupIcd10Parent, searchGroupCpt, searchGroupVisit))
+      .temporal(true)
+      .mention(TemporalMention.ANY_MENTION.name())
+      .time(TemporalTime.WITHIN_X_DAYS_OF.name())
+      .timeValue(5L);
+    searchRequest = new SearchRequest().includes(Arrays.asList(searchGroup));
+    //matches searchGroupCpt and searchGroupVisit
+    assertParticipants(controller.countParticipants(cdrVersion.getCdrVersionId(), searchRequest), 1);
+
+    //First Mention Of DrugChild during same encounter as Measurement
+    //First temporal group
+    searchGroupDrugChild.temporalGroup(0);
+    //Second temporal group
+    searchGroupMeasurement.temporalGroup(1);
+    searchGroup = new SearchGroup()
+      .items(Arrays.asList(searchGroupDrugChild, searchGroupMeasurement))
+      .temporal(true)
+      .mention(TemporalMention.ANY_MENTION.name())
+      .time(TemporalTime.DURING_SAME_ENCOUNTER_AS.name());
+    searchRequest = new SearchRequest().includes(Arrays.asList(searchGroup));
+    //matches searchGroupDrugChild and searchGroupMeasurement
+    assertParticipants(controller.countParticipants(cdrVersion.getCdrVersionId(), searchRequest), 1);
+
+    //Last Mention Of DrugChild during same encounter as Measurement
+    //First temporal group
+    searchGroupDrugChild.temporalGroup(0);
+    //Second temporal group has 2 types(this will envoke the join sql)
+    searchGroupMeasurement.temporalGroup(1);
+    searchGroupVisit.temporalGroup(1);
+    searchGroup = new SearchGroup()
+      .items(Arrays.asList(searchGroupDrugChild, searchGroupMeasurement, searchGroupVisit))
+      .temporal(true)
+      .mention(TemporalMention.LAST_MENTION.name())
+      .time(TemporalTime.DURING_SAME_ENCOUNTER_AS.name());
+    searchRequest = new SearchRequest().includes(Arrays.asList(searchGroup));
+    //matches searchGroupDrugChild and searchGroupMeasurement
+    assertParticipants(controller.countParticipants(cdrVersion.getCdrVersionId(), searchRequest), 1);
+
+    //Last Mention Of (Measurement and visit) X days after DrugParent
+    //First temporal group
+    searchGroupMeasurement.temporalGroup(0);
```

Comment:
These tests are a bit tough to follow. Consider splitting these out into separate test cases, if that runs quickly enough, or possibly splitting some of this out into helper methods or named variables maybe.